File: exercises/D102/solution.pas
var
    dummy: char;
    fare: double;
    tencents, tencentshalved: integer;
begin
    read(dummy, fare);
    tencents := round(fare * 10);
    if (tencents mod 2 = 0) then
        tencentshalved := tencents div 2
    else
        tencentshalved := tencents div 2 + 1;
    writeln('$', tencentshalved / 10 :0:1);
end.

File: exercises/D102/solution.c
#include <stdio.h>
#include <math.h>
int main() {
    char dummy;
    double fare;
    scanf("%c %lf", &dummy, &fare);
    int ten_cents = round(fare * 10);
    int ten_cents_halved;
    if (ten_cents % 2 == 0) {
        ten_cents_halved = ten_cents / 2;
    } else {
        ten_cents_halved = ten_cents / 2 + 1;
    }
    printf("$%.1f\n", ten_cents_halved / 10.0);
    return 0;
}
